crypto/chacha/chacha_enc.c

```diff
@@ -78,10 +79,14 @@ void ChaCha20_ctr32(unsigned char *out, const unsigned char *inp,
     size_t todo, i;
 
     /* sigma constant "expand 32-byte k" in little-endian encoding */
-    input[0] = ((u32)'e') | ((u32)'x'<<8) | ((u32)'p'<<16) | ((u32)'a'<<24);
-    input[1] = ((u32)'n') | ((u32)'d'<<8) | ((u32)' '<<16) | ((u32)'3'<<24);
-    input[2] = ((u32)'2') | ((u32)'-'<<8) | ((u32)'b'<<16) | ((u32)'y'<<24);
-    input[3] = ((u32)'t') | ((u32)'e'<<8) | ((u32)' '<<16) | ((u32)'k'<<24);
+    input[0] = ((u32)os_toascii['e']) | ((u32)os_toascii['x'] << 8)
```

Comment:
Looks like this should be guarded with `#ifdef CHARSET_EBCDIC`

Reply:
This or use ossl_toascii from ctypes.h :+1: 
```
# ifdef CHARSET_EBCDIC
int ossl_toascii(int c);
int ossl_fromascii(int c);
# else
#  define ossl_toascii(c)       (c)
#  define ossl_fromascii(c)     (c)
# endif
```
Not sure whats preferred. The other places in the code all seem to use the ifdef approach ..

Reply:
Actually I didn't know about `ossl_toascii`. This seems a saner approach to me, rather than having two copies of everything.

Reply:
I agree. Id also prefer it over the ifdef.